---
version: "2.1"
services:
  unifi-network-application:
    image: "lscr.io/linuxserver/unifi-network-application:8.0.28-ls27"
    container_name: unifi-network-application
    environment:
      PUID: 1000
      PGID: 1000
      TZ: Etc/UTC
      MONGO_USER: unifiuser
      MONGO_PASS: "{{ lookup('keepass', 'unifi-mongodb-pass', 'password') }}"
      MONGO_HOST: unifi-db
      MONGO_PORT: 27017
      MONGO_DBNAME: unifidb
      MEM_LIMIT: 1024 #optional
      MEM_STARTUP: 1024 #optional
    volumes:
      - unifi-data:/config
    ports:
      - 8443:8443
      - 3478:3478/udp
      - 10001:10001/udp
      - 8080:8080
      - 1900:1900/udp #optional
      - 8843:8843 #optional
      - 8880:8880 #optional
      - 6789:6789 #optional
      - 5514:5514/udp #optional
    restart: always
    networks:
      - mail-relay
      - unifi-internal
    healthcheck:
      test: ["CMD", "curl", "-f", "--insecure", "https://localhost:8443"]
      interval: 30s
      timeout: 10s
      retries: 3

  unifi-db:
    # Starte Container OHNE init-script
    # In Container
    # 1. mongosh
    # 2. db.getSiblingDB("unifidb").createUser({user: "unifiuser", pwd: "GEHEIM", roles: [{role: "dbOwner", db: "unifidb"}, {role: "dbOwner", db: "unifidb_stat"}]});
    # https://discourse.linuxserver.io/t/cant-connect-to-mongodb-for-unifi-network-application/8166
    image: "docker.io/mongo:7.0.5"
    container_name: unifi-db
    volumes:
      - db-data:/data/db
    restart: always
    environment:
      MARIADB_AUTO_UPGRADE: "1"
    networks:
      - unifi-internal
    healthcheck:
      test: ["CMD", "mongosh", "--eval", "db.stats().ok"]
      interval: 30s
      timeout: 10s
      retries: 3

######## Volumes ########
volumes:
  db-data:
  unifi-data:
######## Networks ########
networks:
  mail-relay:
    external: true
  unifi-internal:
    driver: bridge
